src/build_tools.md

```diff
@@ -0,0 +1,122 @@
+# Build Tools
+
+This section covers "build-time" tooling, or code that is run prior to compiling a crate's source code.
+Conventionally, build-time code lives in a `build.rs` file and is commonly referred to as a "build script".
+Common use cases include `rust` code generation and compilation of bundled `C/C++/asm` code.
+See `crates.io`'s [documentation on the matter][build-script-docs] for more information.
+
+Notes:
+- Examples in this section will likely show content for several files.
+- Since [`play.rust-lang.org`][playground] doesn't support specifying/running build scripts, most of
+  the following examples will fail when trying to run them directly in the browser with the
+  "play" (<i class="fa fa-play"></i>) button. Running will require setting up a local project
+  and making the modifications shown.
+
+| Recipe | Crates | Categories |
+|--------|--------|------------|
+| [Compile and link statically to a bundled C library][ex-cc-static-bundled] | [![cc-badge]][cc] | [![cat-development-tools-badge]][cat-development-tools] [![cat-rust-patterns-badge]][cat-rust-patterns] [![cat-external-ffi-bindings-badge]][cat-external-ffi-bindings] |
+
+
+[ex-cc-static-bundled]: #ex-cc-static-bundled
+<a name="ex-cc-static-bundled"></a>
+## Compile and link statically to a bundled C library
+
+[![cat-development-tools-badge]][cat-development-tools] [![cc-badge]][cc] [![cat-rust-patterns-badge]][cat-rust-patterns] [![cat-external-ffi-bindings-badge]][cat-external-ffi-bindings]
+
+To accommodate scenarios where additional `C`, `C++`, or `assembly` is required in a project, the [`cc`][cc] crate
+offers a simple api for compiling bundled `C/C++/asm` code into static libraries (`.a`) that can be statically linked to by `rustc`.
+
+### `Cargo.toml`
+
+```toml
+[package]
+...
+build = "build.rs"
+
+[build-dependencies]
+cc = "1"
+
+[dependencies]
+error-chain = "0.11"
+```
+
+### `build.rs`
+
+```rust,no_run
+extern crate cc;
+
+fn main() {
+    // builds a static lib: `libhello.a`
+    cc::Build::new()
```

Comment:
I'd love to see some hyperlinked references pointing to the used structs and methods as this is the firs formative example in this section.